tools/autograd/gen_variable_type.py

```diff
@@ -185,17 +197,28 @@ def record_trace_outputs(declaration):
 
 def format_trace(declaration):
     local = {}
+    local['trace_name'] = trace_name = uninplace_api_name(declaration['api_name'])
+
```

Comment:
Can there be an example comment here about what this transform is doing. It requires a lot of by-convention knowledge to know that the out argument is first, and that there is an equivalent non-out variant but that it has tensor options.

Reply:
I have the `TensorOptions` thing explained below (where it's handled). I will add the `_out` calling convention comment